File: ch25p731button/p617p633button/MyShrinkingButton.m

#import "MyShrinkingButton.h"


@implementation MyShrinkingButton

- (CGRect)backgroundRectForBounds:(CGRect)bounds {
    CGRect result = [super backgroundRectForBounds:bounds];
    if (self.highlighted)
        result = CGRectInset(result, 3, 3);
    return result;
}


@end

File: iOS8bookExamples/bk2ch12p595button/ch25p866button/MyShrinkingButton.swift


import UIKit

extension CGSize {
    func sizeByDelta(#dw:CGFloat, dh:CGFloat) -> CGSize {
        return CGSizeMake(self.width + dw, self.height + dh)
    }
}

class MyShrinkingButton: UIButton {

    override func backgroundRectForBounds(bounds: CGRect) -> CGRect {
        var result = super.backgroundRectForBounds(bounds)
        if self.highlighted {
            result.inset(dx: 3, dy: 3)
        }
        return result
    }
    
    override func intrinsicContentSize() -> CGSize {
        return super.intrinsicContentSize().sizeByDelta(dw:25, dh: 20)
    }

    
}
